### PR TITLE
NOISSUE - Add linter config file and add duplicate import check

### PR DESCRIPTION
**Pull request by SammyOina**
### What does this do?
- Use config file for linter
- Add checks for duplicate imports using gocritic 
image

### Which issue(s) does this PR fix/relate to?
improves on https://github.com/mainflux/mainflux/issues/1774

### Notes
https://github.com/go-critic/go-critic
https://golangci-lint.run/usage/configuration/

### CODE REVIEW

File: scripts/ci.sh [hunk 1]
```diff
@@ -73,7 +73,7 @@ setup_mf() {
 setup_lint() {
     # binary will be $(go env GOBIN)/golangci-lint
     curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOBIN) $GOLANGCI_LINT_VERSION
-
```
Comment (rodneyosodo) [inline scripts/ci.sh:76]: What about this https://golangci-lint.run/usage/linters/#gocritic ?

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: scripts/ci.sh (hunk 1 above)
Comment (SammyOina) [reply scripts/ci.sh:76]: we can't enable a specific check via CLI

Comment (SammyOina) [reply scripts/ci.sh:76]: switched to a config file in latest

**Review by dborovcanin — CHANGES_REQUESTED** [review]
Please fix the CI.

**Review by dborovcanin — APPROVED** [review]

**Review by rodneyosodo — APPROVED** [review]
LGTM

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**